### DISCUSSION TITLE
Using  KubernetesClient in Spring application

### DISCUSSION THREAD

**Discussion by tinystorm** [fabric8io/kubernetes-client#4449] [Q&A]
I define global client as a bean and using in application runtime.
If i didn't close the client explicitly when  the application is killed or restarted, will there be a connection leak?

**Comment by manusa** [comment]
That depends on the underlying HttpClient you're using (OkHttp by default), but it's very likely.

Usually when declaring the `KubernetesClient` instance as a Bean, Spring should automatically take care of closing it, since it implements the the `Closeable` interface.

In any case, depending on how you're defining the Bean, you can use any of the following methods to hook to the application shutdown callbacks and explicitly call the `KubernetesClient#close` method:
- The Bean definition implements the [`DisposableBean`](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/beans/factory/DisposableBean.html) interface (although I don't think you're defining the bean as a `@Component`)
- Using the `@PreDestroy` `javax` annotation in a method to close the `KubernetesClient`
- Using the `@Bean` annotation [`destroyMethod`](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/context/annotation/Bean.html#destroyMethod--) element to reference the close method.
